### PR TITLE
bitcoin 0.29.x -> 0.30.0

### PR DESCRIPTION
**Pull request by RCasatta**
~~I temporarily added two crates that must be removed before removing the draft status:~~

* ~~hex crate: because I couldn't quickly figure out how to convert bytes to hex~~
* ~~rand: because it didn't seem to be re-exported anymore~~

about `Address`:
* when parameters:  `Address<NetworkChecked>`
* when return values:  `Address<NetworkUnchecked>`

and updated tests with `assume_checked()` where needed

### CODE REVIEW

**Comment by apoelstra** [comment]
@RCasatta you can add `bitcoin_hashes` 0.11 to get `hex`, or depend on `bitcoin-private` which has `DisplayHex`.

`rand` you should be able to get as `bitcoin::secp256k1::rand` ... but can you file a bug if we used to re-export it directly?

**Comment by RCasatta** [comment]
> @RCasatta you can add bitcoin_hashes 0.11 to get hex, or depend on bitcoin-private which has DisplayHex.

I went depending on bitcoin-private because it should be already in the tree

> rand you should be able to get as bitcoin::secp256k1::rand ... but can you file a bug if we used to re-export it directly?

I spent some time figuring out why, it is still exported but the method is missing because `rand-std` is not active by default. Here I solved adding the feature `rand-std` to the rust-bitcoin import but maybe we should add it also  [there](https://github.com/rust-bitcoin/rust-bitcoin/blob/d8c7cd1e6ccbe660722e085f2154fd92fa7aea3b/bitcoin/Cargo.toml#L29) ?

**Comment by apoelstra** [comment]
@RCasatta to get `secp256k1/rand-std` you need to enable `rand-std` on `bitcoin`. We cannot make `secp256k1/rand-std` just be a consequence of `bitcoin/rand` because `bitcoin/rand` does not imply `bitcoin/std`. (It looks like we **did** do this in 0.29.0, which is what changed, but this was a bug and made it impossible to use `rand` and `no-std` on rust-bitcoin.)

File: json/src/lib.rs [hunk 1]
```diff
@@ -1580,13 +1583,13 @@ pub struct GetBlockTemplateResult {
     /// List of transactions included in the template block
     pub transactions: Vec<GetBlockTemplateResultTransaction>,
     /// The signet challenge. Only set if mining on a signet, otherwise empty
-    #[serde(default, with = "bitcoin::blockdata::script::Script")]
-    pub signet_challenge: bitcoin::blockdata::script::Script,
+    #[serde(default, with = "bitcoin::blockdata::script::ScriptBuf")]
```
Comment (tcharding) [inline json/src/lib.rs:1586]: Everything in `blockdata` is re-exported at the crate root so no need for `::blockdata` here or anywhere in this repo.

Reason, incase its not widely known, is that we are trying to separate the API from the directory structure and "blockdata" is an code/module organisation bucket not an API thing.

**Review by tcharding — COMMENTED** [review]
Sick bro! Thanks for doing this so fast, you the man. I would like to see the `blockdata::` thing fixed if possible please, mainly so as to lead downstream users in the right direction.

File: json/src/lib.rs (hunk 1 above)
Comment (RCasatta) [reply json/src/lib.rs]: sure, `::blockdata` removed

**Review by apoelstra — APPROVED** [review]
ACK 201914e74b0d1fc8528093467a2daffcaeabafc3

**Pull request merged**